src/crypto/CHIPCryptoPAL.h

```diff
@@ -1110,11 +1125,29 @@ class Spake2p_P256_SHA256_HKDF_HMAC : public Spake2p
     Spake2pOpaqueContext mSpake2pContext;
 };
 
-/** @brief Clears the first `len` bytes of memory area `buf`.
- * @param buf Pointer to a memory buffer holding secret data that should be cleared.
+/**
+ * @brief Compute the compressed fabric identifier used for operational discovery service
+ *        records from a Node's root public key and Fabric ID. On success, out_compressed_fabric_id
+ *        will have a size of exactly kCompressedFabricIdentifierSize.
+ *
+ * Errors are:
+ *   - CHIP_ERROR_INVALID_ARGUMENT if root_public_key is invalid
+ *   - CHIP_ERROR_BUFFER_TOO_SMALL if out_compressed_fabric_id is too small for serialization
+ *   - CHIP_ERROR_INTERNAL on any unexpected crypto or data conversion errors.
+ *
+ * @param[in] root_public_key The root public key associated with the node's fabric
+ * @param[in] fabric_id The fabric ID associated with the node's fabric
+ * @param[out] out_compressed_fabric_id Span where output will be written. Its size must be >= kCompressedFabricIdentifierSize.
+ * @returns a CHIP_ERROR (see above) on failure or CHIP_NO_ERROR otherwise.
+ */
+CHIP_ERROR GenerateCompressedFabricId(const Crypto::P256PublicKey & root_public_key, uint64_t fabric_id,
```

Comment:
Should `fabric_id` be of type `FabricId`?

Reply:
This is a crypto primitive that is well defined. It would add coupling to whatever header has fabricId to CHIPCryptoPAL. Let me check some alternatives

Reply:
So, FabricId is in `<app/util/basic-types.h>`. I am concerned of coupling between core protocol bits and the crypto primitives which are easily free-standing now. Furthermore, this implements a crypto primitive that is well defined in spec, so I am loathe to tie to other parts of the SDK.

I suggest we revisit this particular type change to when we start using `GenerateCompressedFabricId` in operational discovery.

Reply:
Works for me.  We should move FabricId to a separate header....